### PR TITLE
Use match instead of BASH_REMATCH when using zsh

### LINKED ISSUE #893: Bug: `use` command no longer working as expected with zsh
**Bug report**
With 5.11.1, setting the default for a candidate (i.e. grails) works.  However, if you change the version using 'sdk use', then the candidate can't be found.  I confirmed this on a fresh install with 5.11.1 and on an upgrade.

**To reproduce**
* curl -s https://get.sdkman.io | bash
* open new terminal
* sdk install grails 4.0.9
* which grails <-- the default will succeed / grails can be found until sdk use is called
* sdk use grails 4.0.9
* which grails <-- this will fail

**System info**
* Mac OS Catalina, latest patches
* zsh - zsh 5.7.1 (x86_64-apple-darwin19.0)
* ohmyzsh shell customization with plugins = (git screen mvn jira github gitfast git-flow docker docker-compose kubectl)
* sdk version => SDKMAN 5.11.1+690

### PR DESCRIPTION
**Pull request by helpermethod**
Fixes #893.

### CODE REVIEW

File: src/main/bash/sdkman-use.sh [hunk 1]
```diff
@@ -38,7 +38,10 @@ function __sdk_use() {
 	__sdkman_set_candidate_home "$candidate" "$version"
 
 	if [[ $PATH =~ ${SDKMAN_CANDIDATES_DIR}/${candidate}/([^/]+) ]]; then
-		export PATH=${PATH//${SDKMAN_CANDIDATES_DIR}\/${candidate}\/${BASH_REMATCH[1]}/${SDKMAN_CANDIDATES_DIR}\/${candidate}\/${version}}
+		local matched_version
+		[[ "$zsh_shell" == "true"  ]] && matched_version=${match[1]} || matched_version=${BASH_REMATCH[1]}
```
Comment (marc0der) [inline src/main/bash/sdkman-use.sh:42]: Sorry to be a pain, but do you mind if we break this out into a proper `if` construct for consistency with the rest of the codebase?

Comment (helpermethod) [reply src/main/bash/sdkman-use.sh:42]: Yes, this makes sense.

Comment (helpermethod) [reply src/main/bash/sdkman-use.sh]: Done!

**Comment by helpermethod** [comment]
@marc0der Replaced the "conditional expressions" with proper if-else statements.

**Review by marc0der — APPROVED** [review]
LGTM!

**Pull request merged**